### DISCUSSION TITLE
Achieving different behavior for web and ios/android

### DISCUSSION THREAD

**Discussion by MehmetKaplan** [callstack/react-native-builder-bob#384] [Ideas]
I am trying to implement a library that will behave differently for `web` and `ios` / `android`.

The behavior can be forked at the build time by `react-native`, naming the `js` files as `x.js` and `x.native.js`, where `x.native.js` is used for iOS and Android at the build time, and similarly `x.js` is used for the web at the build time. 

But I am not sure how a library, built with `bob` will behave, considering it is going to be compiled either to CommonJS or ES Modules. For example, if I keep the naming approach same, will it be the same behavior? Any suggestions, how to approach to the subject?

**Brief of the business logic:** Let me share the purpose of the library that I am planning. Even though it is not crucial for the above mentioned question, still it may give an idea, why I need the `web` vs `ios` / `android` behaviors should be different. It will simply open a `WebView` if the environment is the `ios` or `android`. And if the environment is `web` then it will open a new window with `window.open` and drive the user to the newly opened window.

**Comment by satya164** [comment]
It's going to be the same. The files with specific extensions still exist as separate files after compilation - they aren't bundled. So the correct platform will be picked up.

**Comment by MehmetKaplan** [comment]
Hi @satya164 ,

Thanks a lot, as you stated, it is working same way. My above-mentioned library (https://github.com/MehmetKaplan/react-native-webview-with-web) is working for both web and iOS & Android.

So Iet me close the discussion...
